src/Product/AttributeMappingHelper.php

```diff
@@ -0,0 +1,89 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Product;
+
+use Automattic\WooCommerce\GoogleListingsAndAds\Infrastructure\Service;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\Attributes\Adult;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\Attributes\AgeGroup;
+use Automattic\WooCommerce\GoogleListingsAndAds\Product\Attributes\Color;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Helper Class for Attribute Mapping
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Product
+ */
+class AttributeMappingHelper implements Service {
+
+
+	private const ATTRIBUTES_AVAILABLE_FOR_MAPPING = [
+		Adult::class,
+		AgeGroup::class,
+		Color::class,
+	];
+
+	/**
+	 * Gets all the available destinations
+	 *
+	 * @return array
+	 */
+	public function get_destinations(): array {
+		$destinations = [];
+
+		foreach ( self::ATTRIBUTES_AVAILABLE_FOR_MAPPING as $attribute ) {
+			$destinations[ $attribute::get_id() ] = $attribute::get_name();
+		}
+
+		return $destinations;
+	}
+
+	/**
+	 * Gets all the available sources identified by destination key
+	 *
+	 * @return array
+	 */
+	public function get_sources(): array {
+		$sources = [];
+
+		foreach ( self::ATTRIBUTES_AVAILABLE_FOR_MAPPING as $attribute ) {
+			$sources[ $attribute::get_id() ] = $attribute::get_sources();
+		}
+
+		return $sources;
+	}
+
+	/**
+	 * Gets the taxonomies and global attributes to render them as options in the frontend.
+	 *
+	 * @return array An array with the taxonomies and global attributes
+	 */
+	public static function get_source_taxonomies(): array {
+		$taxonomies = get_object_taxonomies( 'product' );
```

Comment:
I'd personally recommend to use `get_taxonomies` here, although it seems to have some issues which is why `get_object_taxonomies` is recommended. The advantage of the `get_taxonomies` call is that a filter can be used. For example product type and product visibility are both non public taxonomies. So I think it would be better to exclude non public ones. It can still be done by comparing the public parameter though.

Reply:
Interesting... when I use get_taxonomies with `public` I miss the product attributes (some of them created by myself)

<img width="501" alt="Screenshot 2022-09-16 at 11 28 12" src="https://user-images.githubusercontent.com/5908855/190581884-6a927fc9-37e0-4c65-a634-007b3640b294.png">

However without `public` it gives me the attributes:

<img width="749" alt="Screenshot 2022-09-16 at 11 26 33" src="https://user-images.githubusercontent.com/5908855/190582054-9403b5ed-d829-493f-9678-768e44807f87.png">

Is there an issue by offering the user the possibility of  using the non public ones as source?

Reply:
Another strange output is that in `get_object_taxonomies` I get some extra taxes like shipping_class or product_visibility

<img width="697" alt="Screenshot 2022-09-16 at 11 34 29" src="https://user-images.githubusercontent.com/5908855/190582978-4b389baa-36cf-47b6-a696-aaf2cc09ab2c.png">

Even though, using `get_taxonomies` (without public attribute set as true) doesn't give me that ones...

Reply:
We have to use `get_object_taxonomies` because of this issue: https://core.trac.wordpress.org/ticket/27918 (note: temporarily unavailable).

But I meant since we are going through the objects in a loop we can continue if `! $taxonomy->public`
It's not a big issue to include public, but I think it returns a cleaner list since I can't think of any specific cases where you would need to use product_type or product_visibility.

Reply:
> But I meant since we are going through the objects in a loop we can continue if ! $taxonomy->public

I tried to do so... but the attributes disappeared.

Reply:
In my testing I added the check for public after the check `taxonomy_is_product_attribute`, so it would still add the product attributes.

But either way my main intent was to filter out product_type and product_visibility. So I'll go ahead and approve this PR and we can tweak later if needed.